linux/ibus-keyman/src/engine.c

```diff
@@ -33,6 +33,10 @@
 #include <X11/XKBlib.h>
 #endif
 
+#ifdef GDK_WINDOWING_WAYLAND
```

Comment:
Can only one display server be active at a time?
Maybe could use
```suggestion
#elif GDK_WINDOWING_WAYLAND
```

Reply:
I would think that it's more about being able to compile in support for multiple display servers for a given platform, as you can switch between them, but being able to disable compiling it in for platforms that don't actually support those display servers.

Reply:
> Can only one display server be active at a time?

Yes, there's only one active at a time, either X11 or Wayland (actually, there's also Wayland with a X11 compatibility layer called XWayland, but that is treated as X11). However, we want to be able to compile binaries that can run in either environment, and not all systems have support for X11 or Wayland therefore these multiple `#ifdef`s instead of `#elif`.